### PR TITLE
Feature/scrollable

### PR DESCRIPTION
**Pull request by dramoturg**

### CODE REVIEW

**Comment by raveclassic** [comment]
merge master

File: package.json [hunk 1]
```diff
@@ -96,6 +96,7 @@
     "react-css-themr": "~1.3.0",
     "react-dom": "^15.3.0",
     "react-event-listener": "^0.2.1",
+    "react-lorem-component": "^0.10.1",
```
Comment (raveclassic) [inline package.json]: why here in deps?

File: src/components/Scrollbar/Bar.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	onBarClick(event) {
+		event.stopPropagation();
+	}
+
+	onBarMouseDown = (event) => {
+		const {onBarDragStart} = this.props;
+
+		onBarDragStart && onBarDragStart(event);
+
+		document.addEventListener('mousemove', this._onDocumentMouseMove);
```
Comment (raveclassic) [inline src/components/Scrollbar/Bar.jsx]: try to use react-event-listener

File: src/components/Scrollbar/HorizontalScrollbar.jsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+		const {container} = this.props;
+		const {height} = this.context.size;
+		container.style.marginBottom = `-${height}px`;
+		container.style.height = `calc(100% + ${height}px)`;
+	}
+
+	componentDidMount() {
+		super.componentDidMount();
+		this._minBarSize = this._getMinBarSize();
+		this._hideNativaScrollContainer();
```
Comment (raveclassic) [inline src/components/Scrollbar/HorizontalScrollbar.jsx]: typo

File: package.json (hunk 1 above)
Comment (dramoturg) [inline package.json]: Remove. This package was used to create the test content scrollable block.

**Comment by raveclassic** [comment]
Dragging behavior is inconsistent: mousedown, drag out of scrollbar, mouseup.
Why do you need the buttons? Hold is not working properly, hover/pressed state is inconsistent.

**Comment by raveclassic** [comment]
there's a slight performance regression on scroll
image

File: src/components/Scrollbar/HorizontalScrollbar.jsx (hunk 3 above)
Comment (raveclassic) [inline src/components/Scrollbar/HorizontalScrollbar.jsx]: Dragging behavior is inconsistent: mousedown, drag out of scrollbar, mouseup.
Why do you need the buttons? Hold is not working properly, hover/pressed state is inconsistent.

**Pull request merged**